src/main/java/org/jboss/forge/arquillian/container/provider/ContainerDirectoryParser.java

```diff
@@ -35,7 +37,20 @@ void createContainers() {
                 new TypeReference<List<Container>>() {
                 });
 
-            containers = Collections.unmodifiableList(parsedContainers);
+            Loader loader = new Loader();
+            final org.arquillian.container.chameleon.spi.model.Container[] containers = loader.loadContainers(Target.class.getClassLoader().getResourceAsStream("chameleon/default/containers.yaml"));
+
+            parsedContainers.stream()
+                .filter(container -> container.getGroupId() == null && container.getArtifactId() == null)
+                .forEach(container -> {
+                    try {
+                        container.setGroupIdAndArtifactIdFromChameleonConfiguration(containers);
+                    } catch (Exception e) {
+                        throw new IllegalStateException("Couldn't set Group Id & Artifact Id for container" + container.getName());
+                    }
+                });
+
+            ContainerDirectoryParser.containers = Collections.unmodifiableList(parsedContainers);
```

Comment:
Any benefit of having it `static`? It's a bit strange in this shape, especially that we have `getter` for it too.